Source/hydro/riemann_solvers.H

```diff
@@ -2,6 +2,664 @@
 #define riemann_solvers_H
 
 #include <Castro_util.H>
+#include <riemann.H>
+
+///
+/// The Colella-Glaz Riemann solver for pure hydrodynamics.  This is a
+/// two shock approximate state Riemann solver.
+///
+/// @param bx         the box to operate over
+/// @param ql         the left interface state
+/// @param qr         the right interface state
+/// @param qaux_arr   the auxillary state
+/// @param qint       the full Godunov state on the interface
+/// @param idir       coordinate direction for the solve (0 = x, 1 = y, 2 = z)
+///
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void
+riemanncg(const int i, const int j, const int k,
+          const RiemannState ql, const RiemannState qr, const RiemannAux raux,
```

Comment:
I think it would be better to pass these by reference, no need to make a copy here

Reply:
fixed